app/jobs/search_keyword_job.rb

```diff
@@ -4,11 +4,26 @@ class SearchKeywordJob < ApplicationJob
   queue_as :default
 
   def perform(keyword_id)
-    keyword = Keyword.find(keyword_id)
-    html = GoogleSearchService.new(keyword: keyword.keyword).call
-    utf_8_html = html.force_encoding('iso8859-1').encode('utf-8')
-    keyword.add_html(utf_8_html)
-  rescue ActiveRecord::RecordNotFound, GoogleSearch::Errors::SearchKeywordError, ActiveRecord::StatementInvalid
-    keyword.update_status(:failed)
+    @keyword = Keyword.find(keyword_id)
+    begin
+      html = GoogleSearchService.new(keyword: keyword.keyword).call
+      parse_result = GoogleParseService.new(html: html).result
+      save_data(html, parse_result)
+    rescue ActiveRecord::RecordNotFound, GoogleSearch::Errors::SearchKeywordError, ActiveRecord::StatementInvalid
+      keyword.update_status(:failed)
+    end
+  end
+
+  private
+
+  attr_reader :keyword
+
+  def save_data(html, parse_result)
```

Comment:
This function name is quite ambiguous, how about `save_keyword_results`?

Reply:
Updated the name! Good catch!
Fixed in [`5f004ca` (#47)](https://github.com/suho/google-search-ruby/pull/47/commits/5f004cae025caf7b86ce910239bd5478850fb86a)